data/transactions/logic/eval.go

```diff
@@ -255,6 +258,9 @@ func (ep EvalParams) budget() int {
 	if ep.runModeFlags == runModeSignature {
 		return int(ep.Proto.LogicSigMaxCost)
 	}
+	if ep.Proto.EnableAppCostPooling && ep.PooledApplicationBudget != nil {
```

Comment:
Is the `nil` check necessary here? It fails some tests, e.g. `ledger/apply/application_test.go:TestAppCallApplyCreateOptIn()`, without it. I thought `prepareEvalParams()` was always called before an App call, and if this is the case, can we remove this `nil` check and modify those unit tests?

Reply:
yes, please modify the test. It constructs `var ep logic.EvalParams` directly since there is no access to `prepareEvalParams` from `apply` package